format/FlightSql.proto

```diff
@@ -760,6 +786,20 @@ enum SqlInfo {
   SQL_STORED_FUNCTIONS_USING_CALL_SYNTAX_SUPPORTED = 576;
 }
 
+// The level of support for Flight SQL transaction RPCs.
+enum FlightSqlTransactionSupport {
+  // Unknown/not indicated
+  FLIGHT_SQL_TRANSACTION_SUPPORT_UNKNOWN = 0;
+  // No support
+  FLIGHT_SQL_TRANSACTION_SUPPORT_NONE = 1;
+  // Transactions, but not savepoints.
+  // A savepoint is a mark within a transaction that can be individually
+  // rolled back to. Not all databases support savepoints.
```

Comment:
Just for my understanding, savepoints are for two-phase commits, right?

Reply:
Hmm, not in the distributed systems sense, a savepoint is really just a nested transaction. I included it for parity with JDBC but it's not essential.